### PR TITLE
Allow users to set status code 0 in Response to maintain backward compatibility in OkHttp4.0

### PR DESCRIPTION
**Pull request by monkey-mas**
Fixes https://github.com/square/okhttp/issues/5209

Previously in OkHttp3, status code == 0 was acceptable.
ref. [status code validity check in 3.14.2](https://github.com/square/okhttp/blob/parent-3.14.2/okhttp/src/main/java/okhttp3/Response.java#L453)

### CODE REVIEW

**Comment by JakeWharton** [comment]
Test to prevent regression?

**Comment by monkey-mas** [comment]
> tests
Okay will do :)

File: okhttp/src/main/java/okhttp3/Response.kt [hunk 1]
```diff
@@ -433,7 +433,7 @@ class Response internal constructor(
     }
 
     open fun build(): Response {
-      check(code > 0) { "code < 0: $code" }
+      check(code >= 0) { "code < 0: $code" }
```
Comment (monkey-mas) [inline okhttp/src/main/java/okhttp3/Response.kt]: nit: I kinda feel this should probably throw `IllegalArgumentException`? Anyways, this is unrelated to backward compatibility. Let's discuss it in another issue/pr :)

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/Response.kt]: The function accepts no arguments though

Comment (monkey-mas) [reply okhttp/src/main/java/okhttp3/Response.kt]: ah, correct..  Okay, so if `Builder#code(...)` threw an exception, the exception could be `IllegalArgumentException `, which Builder doesn't do. In our case, `Builder#build()` uses _state_ that was set via setters and throws `IllegalStateException`.

Hope my understanding is correct, thanks anyway :)
